plugins/hydra_ax_sweeper/hydra_plugins/hydra_ax_sweeper/config.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import MISSING, ObjectConf
+
+
+@dataclass
+class EarlyStopConfig:
+
+    minimize: bool = MISSING
+
+    max_epochs_without_improvement: int = 10
+
+    # An improvement larger than epsilon is considered significant
+    epsilon: float = 0.00001
+
+
+@dataclass
+class ExperimentConfig:
+
+    # Experiment name
+    name: Optional[str] = None
+
+    # Name of metric to optimize or null if you only return one metric
+    objective_name: str = "objective"
+
+    # Defaults to minimize, set to false to maximize
+    minimize: bool = True
+
+    # For the remaining paramaters, refer the Ax documentation: https://ax.dev/api/core.html#experiment
+    parameter_constraints: Optional[List[str]] = None
+    outcome_constraints: Optional[List[str]] = None
+    status_quo: Optional[Dict[str, Any]] = None
+
+
+@dataclass
+class ClientConfig:
+
+    verbose_logging: bool = False
+
+    # set random seed here to make Ax results reproducible
+    random_seed: Optional[int] = None
+
+
+@dataclass
+class AxConfig:
+
+    # max_trials is application-specific. Tune it for your use case
+    max_trials: int = 10
+
+    early_stop: EarlyStopConfig = EarlyStopConfig()
+
+    experiment: ExperimentConfig = ExperimentConfig()
+
+    client: ClientConfig = ClientConfig()
+
+    params: Dict[str, Any] = field(default_factory=dict)
```

Comment:
Be more conservative with new lines between members. add them to separate distinct groups.
If there is only one group don't add newlines in between everything.